source/agora/test/Base.d

```diff
@@ -1028,3 +1034,79 @@ public bool containSameBlocks (APIS)(APIS nodes, size_t height)
 
     return true;
 }
+
+
+/*******************************************************************************
+
+    Generate a genesis block.
+
+    For each key pair, a freeze transaction and an Enrollment
+    will be created and added to the generated Genesis block.
+    The first transaction is the `UnitTestGenesisTransaction`
+    The minimum number of enrollments is Enrollment.MinValidatorCount.
+
+    Params:
+        key_pairs = key pairs for signing enrollments with
+
+    Returns:
+        The genesis block
+
+*******************************************************************************/
+
+private immutable(Block) makeGenesisBlock (in KeyPair[] key_pairs)
+{
+    import agora.common.Serializer;
+
+    // 1 payment tx, the rest are freeze txs
+    Transaction[] txs;
+    txs ~= UnitTestGenesisTransaction.serializeFull.deserializeFull!Transaction;
+    Enrollment[] enrolls;
+
+    foreach (idx; 0 .. max(Enrollment.MinValidatorCount, key_pairs.length))
```

Comment:
It would be much better to add an assertion that `key_pairs.length >= Enrollment.MinValidatorCount`.

If you generate random keypairs here, how will they be used by the Validators? Who will validate blocks [1..$]?

Reply:
Only one validator is created when the value of TestConf.topology is NetworkTopology.OneValidator.
This is because the existing test code does not work.
So I created a dummy enrollment data. Is there any other good way to solve this?

Reply:
Just let it be one validator. There don't have to be two validators.

Reply:
Thank you for the good suggestion.